baseplate_test.go

```diff
@@ -145,6 +150,72 @@ func TestServe(t *testing.T) {
 	}
 }
 
+type timestampCloser struct {
+	ts []time.Time
+}
+
+func (c *timestampCloser) Close() error {
+	c.ts = append(c.ts, time.Now())
+	return nil
+}
+
+func TestServeClosers(t *testing.T) {
+	t.Parallel()
+
+	store := newSecretsStore(t)
+	defer store.Close()
+
+	bp := baseplate.NewTestBaseplate(baseplate.Config{StopTimeout: testTimeout}, store)
+
+	pre := &timestampCloser{}
+	post := &timestampCloser{}
+
+	args := baseplate.ServeArgs{
+		Server:       newWaitServer(t, bp, time.Millisecond),
+		PreShutdown:  []io.Closer{pre},
+		PostShutdown: []io.Closer{post},
+	}
+
+	t.Run(
+		"serve closer order",
+		func(t *testing.T) {
+			ch := make(chan error)
+
+			go func() {
+				// Run Serve in a goroutine since it is blocking
+				ch <- baseplate.Serve(
+					context.Background(),
+					args,
+				)
```

Comment:
2 style nits:

1. This should be fine to fit on one line...
2. Modulo that you should just define `args` inline here

```suggestion
				ch <- baseplate.Serve(context.Background(), baseplate.ServeArgs{
					Server:       newWaitServer(t, bp, time.Millisecond),
					PreShutdown:  []io.Closer{pre},
					PostShutdown: []io.Closer{post},
				})
```